tests/unittests/Foundation/NSUndoManagerTests.mm

```diff
@@ -0,0 +1,239 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <Foundation/Foundation.h>
+#import <TestFramework.h>
+
+@interface someObject : NSObject
+@property NSUndoManager* undoManager;
+@property NSNumber* value;
+@property NSMutableArray* arrayOfNumbers;
+- (id)initWithValue:(NSNumber*)num;
+- (id)initWithEmptyArray;
+- (void)updateNumber:(NSNumber*)value;
+- (void)updateNumberSingleUndoStep:(NSNumber*)value;
+- (void)addToMutableArray:(NSNumber*)value;
+- (void)removeFromMutableArray:(NSNumber*)value;
+@end
+
+@implementation someObject
+
+- (id)initWithValue:(NSNumber*)num {
+    self = [super init];
+    _undoManager = [[[NSUndoManager alloc] init] autorelease];
```

Comment:
>autorelease [](start = 49, length = 11)

There is no retain when you assign to _undoManager, so autorelease should not be there.